aws_signing_helper/pkcs11_signer.go

```diff
@@ -208,16 +469,30 @@ func (pkcs11Signer *PKCS11Signer) Sign(rand io.Reader, digest []byte, opts crypt
 		return nil, fmt.Errorf("signing initiation failed (%s)", err.Error())
 	}
 
-	err = module.Login(session, pkcs11.CKU_CONTEXT_SPECIFIC, pkcs11Signer.pin)
-	if err != nil {
-		return nil, fmt.Errorf("user re-authentication failed (%s)", err.Error())
+	// We assume it's the same PIN as the token itself? Which was only
+	// "saved" for us in pkcs11Signer if the CKA_ALWAYS_AUTHENTICATE
```

Comment:
Maybe I'm misunderstanding what `CKA_ALWAYS_AUTHENTICATE` should be doing, but I thought it meant that we should prompt the user for their PIN each time we want to do a signing operation. Is this not the case?

Reply:
Whether we prompt the *user* for it is our decision. We do need to provide it to the module each time.

I think that in *theory* it might be possible for the `CKU_CONTEXT_SPECIFIC_PIN` for a given key to be *different* to the `CKU_USER` pin, and we're using the same user input for both?

I'd actually like to do this with a *callback* for the user interaction, and we can invoke it each time we need a (new) PIN.